### DISCUSSION TITLE
General QoL and polish before 1.0

### DISCUSSION THREAD

**Discussion by jwoertink** [luckyframework/lucky#1563] [Feature requests & Ideas]
I keep thinking about this, but need a place to write it down. This is a list of the items I'd like to clean up for a general "quality of life" and overall polish for the next release. If you think of anything you'd like to see, please feel free to add it in.

These aren't necessarily features, or bug fixes, but more just things to clean up the UI/UX of working with Lucky.

* [ ] - Some compiling notices use the spinner, some don't. Make them all use a spinner
* [x] - Running `lucky routes` used to look pretty, then I ruined it 😬  I want to revert that and use the shell table again
* [ ] - How can we get tab autocomplete on lucky tasks? That would be slick
* [ ] - Make all task usage failures consistent. Ref: https://github.com/luckyframework/lucky/issues/1463 
* [ ] - We will eventually have our own process manager... This will need to look pretty
* [ ] - Make exceptions and compile-time errors consistent. They should all follow a guide of how they read and look
* [x] - The `db.migrations.status` output should show "Migrated" as green, and "Pending" as yellow so it stands out a little more

**Comment by jkthorne** [comment]
can we integrate with docker or docker compose instead of using a process manager?

**Reply by jwoertink** [reply]
It wouldn't be bad to make things run through Docker natively; however, I absolutely can't stand Docker.... I always have so many [issues](https://github.com/docker/for-mac/issues/5677#issuecomment-940347181) with it, so for me personally, I would need a way to not run in Docker.

But maybe we can find a nice way to set your process so when you run `lucky dev`, if you selected Docker, it would boot all in that. Probably not a "before 1.0" kind of thing, but I like the idea of having an option.

**Reply by jwoertink** [reply]
One thing I have thought about adding was the concept of a `.luckyrc` type file... Like setting a global editor for `lucky exec`, and whatever else... Maybe this is where that would go?

```
# ~/.luckyrc
EDITOR=emacs
PROCESS_MANAGER=docker
NODE_PACKAGE_MANAGER=npm
```

no clue.. just tossing out some ideas.

**Reply by jkthorne** [reply]
hmm maybe docker is not the best then.  It is a very common tool and I have been pushed to it more and more on projects so it should probably be supported in some way.

**Reply by jwoertink** [reply]
Agreed. There's https://github.com/luckyframework/lucky_cli/issues/229 for adding the config in, but maybe we just always generate a simple one... You don't have to use it, but it would be there as a jumping-off point at least.
